src/schemas/m2cms.ts

```diff
@@ -0,0 +1,152 @@
+import process from 'process';
+import fs from 'node:fs';
+import path from 'node:path';
+import Typescript, {
+    type Node,
+    type InterfaceDeclaration,
+    type PropertySignature,
+    type Identifier
+} from 'typescript';
+import YAML from 'yaml';
+
+import Debug from 'debug';
+const DEBUG = Debug('m2cms')
+
+const KIND = (e: Node | undefined ) => e && Typescript.SyntaxKind[e.kind]
+
+interface Field {
+    label: string;
+    name: string;
+    widget: string;
+    fields?: Field[];
+}
```

Comment:
this is a 'field' in DecapCMS terminology,
it is basically a line in the config.

Reply:
Mildly unclear to me, maybe linking to the section of the DecapCMS documentation that describes this model would make it easier to read.